openff/qcsubmit/_tests/test_submissions.py

```diff
@@ -835,6 +838,139 @@ def test_optimization_submissions(fulltest_client, specification):
                 assert "scf quadrupole" in result.properties.keys()
 
 
+@pytest.mark.parametrize(
+    "opt_keywords",
+    [
+        pytest.param(
+            (
+                "CUSTOM",
+                [
+                    "energy",
+                    "1e-8",
+                    "maxiter",
+                ],
+                3,
+                "custom convergence with maxiter",
+            ),
+            id="Custom convergence with maxiter",
+        ),
+        pytest.param(
+            (
+                "GAU_VERYTIGHT",
+                [
+                    "maxiter",
+                ],
+                3,
+                "Default conv with maxiter",
+            ),
+            id="Default conv with maxiter",
+        ),
+        pytest.param(
+            (
+                "CUSTOM",
+                [
+                    "energy",
+                    "1e-4",
+                    "grms",
+                    "3e-2",
+                    "gmax",
+                    "4.5e-1",
+                    "drms",
+                    "1.2e-3",
+                    "dmax",
+                    "1.8e-1",
+                ],
+                300,
+                "Custom convergence, no maxiter",
+            ),
+            id="Custom convergence, no maxiter",
+        ),
+    ],
+)
+def test_optimization_submissions_convergence(fulltest_client, opt_keywords):
+    """Test submitting an Optimization dataset with custom convergence options."""
+
+    client = fulltest_client
+
+    convergence_set, converge, maxit, ds_suffix = opt_keywords
+
+    ethane = Molecule.from_file(get_data("ethane.sdf"), "sdf")
+
+    dataset = OptimizationDataset(
+        dataset_name="Test optimizations with converge " + ds_suffix,
+        description="Test optimization dataset with constraints" + ds_suffix,
+        dataset_tagline="Testing optimization datasets" + ds_suffix,
+    )
+
+    dataset.clear_qcspecs()
+    dataset.add_qc_spec(
+        method="openff-1.0.0",
+        basis="smirnoff",
+        program="openmm",
+        spec_name="test_spec",
+        spec_description="test_spec",
+        overwrite=True,
+    )
+
+    # add the molecule
+    index = ethane.to_smiles()
+    dataset.add_molecule(index=index, molecule=ethane)
+
+    # force a validation error with the GeometricProcedure
+    with pytest.raises(ValidationError):
+        dataset.optimization_procedure = GeometricProcedure(
+            convergence_set="GAU maxiter"
+        )
+    with pytest.raises(ValidationError):
+        dataset.optimization_procedure = GeometricProcedure(converge=["GAU", "maxiter"])
```

Comment:
It might be best to move these to their own tests, that should make it easier to work out what has happened if this test failed as currently it could be caused by three different cases.